Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm

```diff
@@ -59,11 +60,208 @@
+    if ([attribute isEqualToString:NSAccessibilityTopLevelUIElementAttribute])
+        return [[self parent] accessibilityAttributeValue:NSAccessibilityTopLevelUIElementAttribute];
+    if ([attribute isEqualToString:NSAccessibilityWindowAttribute])
+        return [[self parent] accessibilityAttributeValue:NSAccessibilityWindowAttribute];
+    if ([attribute isEqualToString:NSAccessibilitySizeAttribute])
+        return [NSValue valueWithSize:_pdfPlugin->boundsOnScreen().size()];
+    if ([attribute isEqualToString:NSAccessibilityEnabledAttribute])
+        return [[self parent] accessibilityAttributeValue:NSAccessibilityEnabledAttribute];
+    if ([attribute isEqualToString:NSAccessibilityPositionAttribute])
+        return [NSValue valueWithPoint:_pdfPlugin->boundsOnScreen().location()];
+    if ([attribute isEqualToString:NSAccessibilityRoleAttribute])
+        return NSAccessibilityGroupRole;
+    if ([attribute isEqualToString:NSAccessibilityPrimaryScreenHeightAttribute])
+        return [[self parent] accessibilityAttributeValue:NSAccessibilityPrimaryScreenHeightAttribute];
+    if ([attribute isEqualToString:NSAccessibilitySubroleAttribute])
+        return @"AXPDFPluginSubrole";
+    return nil;
+}
+
+ALLOW_DEPRECATED_IMPLEMENTATIONS_BEGIN
+- (NSArray *)accessibilityAttributeNames
+ALLOW_DEPRECATED_IMPLEMENTATIONS_END
+{
+    static NeverDestroyed<RetainPtr<NSArray>> attributeNames = @[
+        NSAccessibilityParentAttribute,
+        NSAccessibilityWindowAttribute,
+        NSAccessibilityTopLevelUIElementAttribute,
+        NSAccessibilityRoleDescriptionAttribute,
+        NSAccessibilitySizeAttribute,
+        NSAccessibilityEnabledAttribute,
+        NSAccessibilityPositionAttribute,
+        NSAccessibilityFocusedAttribute,
+        NSAccessibilityChildrenAttribute,
+        NSAccessibilityPrimaryScreenHeightAttribute,
+        NSAccessibilitySubroleAttribute
+    ];
+    return attributeNames.get().get();
+}
+
+- (BOOL)accessibilityShouldUseUniqueId
+{
+    return YES;
+}
+
+- (NSUInteger)accessibilityArrayAttributeCount:(NSString *)attribute
+{
+    if (!_pdfDocument)
+        _pdfDocument =_pdfPlugin->pdfDocumentForPrinting();
+    if ([attribute isEqualToString:NSAccessibilityChildrenAttribute])
+        return [_pdfDocument.get() pageCount];
+    if ([attribute isEqualToString:NSAccessibilityVisibleChildrenAttribute])
+        return [self accessibilityVisibleChildren].count;
+    return [super accessibilityArrayAttributeCount:attribute];
+}
+
+- (NSArray*)accessibilityChildren
+{
+    if (!_pdfDocument)
+        _pdfDocument =_pdfPlugin->pdfDocumentForPrinting();
+    return [_pdfDocument.get() accessibilityChildren:self];
+}
+
+- (id)accessibilityHitTest:(NSPoint)point
+{
+    for (id element in [self accessibilityChildren]) {
+        id result = [element accessibilityHitTest:point];
+        if (!result)
+            return result;
+    }
+    return self;
+}
+
+- (void)gotoDestination:(PDFDestination *)destination
```

Comment:
Not sure if you can change the name, but it's really unclear what "goto" means here. If it triggers a scroll, we can use the term "reveal" or "scrollTo".

Reply:
this is called by `- (void) scrollToVisible` function in `PDFAccessibilityNodePage.m` and PDFLayerController use the same function name. I agree it is unclear and I will add a comment!